gui/src/app/SPAnalysis/SPAnalysisContextProvider.tsx

```diff
@@ -34,18 +48,31 @@ const SPAnalysisContextProvider: FunctionComponent<PropsWithChildren<SPAnalysisC
             window.removeEventListener('beforeunload', handleBeforeUnload);
         };
     }, [data])
+    ////////////////////////////////////////////////////////////////////////////////////////
+
 
     useEffect(() => {
-        // load the saved state on first load
-        const savedState = localStorage.getItem('stan-playground-saved-state')
-        if (!savedState) return
-        const parsedData = deserializeAnalysis(savedState)
-        update({ type: 'loadLocalStorage', state: parsedData })
-    }, [])
-    ////////////////////////////////////////////////////////////////////////////////////////
+        if (data != initialDataModel) return;
+
+        const query = fromSearchParams(searchParams);
+
+        // any query is set
+        if (Object.values(query).some(v => v !== null)) {
```

Comment:
Might be worth wrapping this in a semantically-named function `queryStringHasParameters: QueryParams -> boolean`--not super important, it'd still be a one-liner, but it would read a little more obviously than having to comment it (& would let you keep the definition of what a 'meaningful query string' is closer to the code that handles the query strings)